### PR TITLE
fix!: update the fee collector name to match Agoric usage

### PR DESCRIPTION
**Pull request by JimLarson**
## Description

Refs: Agoric/agoric-sdk#9036

Update the fee collector module account name to match Agoric usage.

This eliminates the need for Agoric/agoric-sdk to maintain its own clone of `DeductFeeDecorator`.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] added `!` to the type prefix if API or client breaking change
- [x] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] provided a link to the relevant issue or specification
- [x] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/building-modules)
- [x] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#testing)
- [x] added a changelog entry to `CHANGELOG.md`
- [x] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [x] updated the relevant documentation or specification
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/auth/types/keys.go [hunk 1]
```diff
@@ -12,7 +12,8 @@ const (
 	StoreKey = "acc"
 
 	// FeeCollectorName the root string for the fee collector account address
-	FeeCollectorName = "fee_collector"
+	// Agoric: maintain compatibility with Agoric/agoric-sdk/golang/cosmos/x/vbank/types.ReservePoolName
```
Comment (gibson042) [inline x/auth/types/keys.go:15]:
```suggestion
	// Agoric: collect fees into the reserve pool
	// cf. Agoric/agoric-sdk golang/cosmos/x/vbank/types/key.go `ReservePoolName`
```

**Comment by JimLarson** [comment]
Lint failure is expected.

I noticed that we're not following the `CHANGELOG` convention in `CHANGELOG-Agoric` - we should be adding a link to the PR, not to the issue. This means that you can't have the changelog entry when you create the PR since you need the number.

I always forget to add the changelog entry, so now that seems prescient.

**Review by gibson042 — APPROVED** [review]
LGTM with a few comment suggestions. Is this PR targeting the right branch, or should we create a new one for this and #405?

<pre>

*   a0154d6890 (<strong>Agoric</strong>) Merge pull request #404 from agoric-labs/jimlarson-lint
|\  
| * e96708b346 chore: silence some lint warnings
|/  
*   9c20eacace Merge pull request #385 from agoric-labs/8989-rm-lien
|\  
| * b3b9528ad5 docs: changelog
| * 28ed7db56c chore: restore vesting account LockedCoins signature
| * 41d452862b chore: remove gymnastics around vesting account actions
| * dd6bf90d83 chore: revert interface-ification of AccountKeeper
|/  
*   0d5eab2ce2 Merge pull request #383 from agoric-labs/8890-disable-sims
|\  
| * e592d3a4b0 tests: disable sims which always run too long
|/  
*   f42d86980d (<strong><em>tag:</em> v0.46.16-alpha.agoric.2</strong>) Merge pull request #379 from agoric-labs/mhofman/backport-19371
|\  
| * 84d5de04d3 fix: avoid cli redundant log in stdout (#19371)
* | b0baff1a47 Merge pull request #378 from agoric-labs/8871-submit-proposal
|\| 
| * aa291d837c docs: update changelog
| * 5cec00b6ea feat!: have tx gov submit-proposal accept either new or legacy syntax
|/  
*   c243c2e43a (<strong><em>tag:</em> v0.46.16-alpha.agoric.1</strong>) Merge pull request #376 from agoric-labs/8224-tagnotes

</pre>

**Comment by JimLarson** [comment]
We'll follow the practice of landing changes on trunk first, then cherry-pick to the (to-be-created) release branch.

**Comment by JimLarson** [comment]
#405 should be cherry-picked there too.

Following past precedent, which it's unnecessary to improve upon, we'll branch `Agoric-v0.46.16.2.x` off of `v0.46.16-alpha.agoric.2` and lay down the tag `v0.46.16-alpha.agoric.2.1`.

**Comment by JimLarson** [comment]
Won't work due to other functionality that automatically distributes the fee collector account - closing.

**Pull request closed without merging**